src/server.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import unicode_literals
+import socket
+import os
+import io
+from email.utils import formatdate
+
+class HTTPErrors(Exception):
+    """HTTP Error exception class."""
+
+    def __init__(self, message):
+        """Intialize Error with message."""
+        self.message = message
+
+
+def resolve_uri(uri):
+    """Resolve uri and return body for response."""
+    root = os.path.abspath("./webroot")
+    path = root + uri
+    content_type = u""
+    if os.path.isfile(path):
+        with io.open(path, 'rb') as file:
+            content = file.read()
+        file.close()
+        if uri.endswith('txt'):
+            content_type = u"text"
+        elif uri.endswith('html'):
+            content_type = u"HTML"
+        elif uri.endswith(('png', 'jpg', 'jpeg')):
+            content_type = u"image"
```

Comment:
don't forget the `mimetypes` module, in the Python standard library.  It will do a better job of handling this that you will.